plugins/lookup/consul_kv.py

```diff
@@ -29,13 +29,17 @@
       index:
         description:
           - If the key has a value with the specified index then this is returned allowing access to historical values.
+        type: int
       datacenter:
         description:
           - Retrieve the key from a consul datacenter other than the default for the consul host.
+        type: str
```

Comment:
Just curious why this is 'str' and not 'string' ?

Reply:
We use 'str' to [get the validation function](https://github.com/ansible/ansible/blob/906c969b551b346ef54a2c0b41e04f632b7b73c2/lib/ansible/module_utils/common/parameters.py#L109). `str` is also the class name for constructing string objects in Python, so that's probably why 'str' was used instead of 'string'.

Reply:
Not the correct validator, we still have separate one for plugins, the above one is for modules. 

The str vs string is now just a 'preference' as we updated the plugin config to understand both: https://github.com/ansible/ansible/blob/devel/lib/ansible/config/manager.py#L161

Reply:
its on my list to 'unite' but i have to handle differences, this one is simple, fun one is 'plugins dont handle suboptions'

Reply:
Ah, that's right. I forgot we used separate validatiors for plugins.

Reply:
I used `str` here because `type: str` was already used in this flie (see further below, for the `url` option).

> fun one is 'plugins dont handle suboptions'

That's one I've been waiting for for a long time :) I tend to forget that they don't, and then am surprised when something strange comes through `self.get_option()`...

Reply:
thanks everyone for the replies!